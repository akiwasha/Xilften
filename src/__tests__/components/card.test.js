import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Card, Player } from '../../components';

const category = 'series';

const slideRows = [
  {
    title: 'Documentaries',
    data: [
      {
        title: 'Tiger King',
        description:
          'An exploration of big cat breeding and its bizarre underworld, populated by eccentric characters.',
        genre: 'documentaries',
        maturity: '18',
        slug: 'tiger-king',
      },
    ],
    title: 'Documentaries',
    data: [
      {
        title: 'Amanda Knox',
        description:
          'Amanda Marie Knox is an American woman who spent almost four years in an Italian prison.',
        genre: 'documentaries',
        maturity: '12',
        slug: 'amanda-knox',
      },
    ],
  },
];

describe('<Card />', () => {
  it('renders the <Card /> component with the populated data', () => {
    const { container, getByText } = render(
      <Card.Group>
        {slideRows.map((slideItem) => (
          <Card key={`${category}-${slideItem.title.toLowerCase()}`}>
            <Card.Title>{slideItem.title}</Card.Title>
            <Card.Entities>
              {slideItem.data.map((item, i) => (
                <Card.Item key={i} item={item}>
                  <Card.Image
                    src={`/images/${category}/${item.genre}/${item.slug}/small.jpg`}
                  />
                  <Card.Meta>
                    <Card.SubTitle>{item.title}</Card.SubTitle>
                    <Card.Text>{item.description}</Card.Text>
                  </Card.Meta>
                </Card.Item>
              ))}
            </Card.Entities>
            <Card.Feature category={category}>
              <Player>
                <Player.Button />
                <Player.Video src="/videos/bunny.mp4" />
              </Player>
            </Card.Feature>
          </Card>
        ))}
      </Card.Group>
    );

    expect(getByText('Documentaries')).toBeTruthy();
    expect(getByText('Amanda Knox')).toBeTruthy();
    expect(
      getByText(
        'Amanda Marie Knox is an American woman who spent almost four years in an Italian prison.'
      )
    ).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders the <Card /> and toggles the card feature', () => {
    const { container, queryByText, getByTestId, getByAltText } = render(
      <Card.Group>
        {slideRows.map((slideItem) => (
          <Card key={`${category}-${slideItem.title.toLowerCase()}`}>
            <Card.Title>{slideItem.title}</Card.Title>
            <Card.Entities>
              {slideItem.data.map((item, i) => (
                <Card.Item
                  key={i}
                  item={item}
                  data-testid={`${item.slug}-item-feature`}
                >
                  <Card.Image
                    src={`/images/${category}/${item.genre}/${item.slug}/small.jpg`}
                  />
                  <Card.Meta>
                    <Card.SubTitle>{item.title}</Card.SubTitle>
                    <Card.Text>{item.description}</Card.Text>
                  </Card.Meta>
                </Card.Item>
              ))}
            </Card.Entities>
            <Card.Feature category={category}>
              <Player>
                <Player.Button />
                <Player.Video src="/videos/bunny.mp4" />
              </Player>
            </Card.Feature>
          </Card>
        ))}
      </Card.Group>
    );

    expect(queryByText('12')).toBeFalsy();
    fireEvent.click(getByTestId('amanda-knox-item-feature'));
    expect(queryByText('12')).toBeTruthy();

    fireEvent.click(getByAltText('Close'));
    expect(queryByText('12')).toBeFalsy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
