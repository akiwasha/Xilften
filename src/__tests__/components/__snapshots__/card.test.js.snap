// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders the <Card /> and toggles the card feature 1`] = `
<div
  class="card__Group-bu7s43-2 kghlzN"
>
  <div
    class="card__Container-bu7s43-1 gXEVLR"
  >
    <p
      class="card__Title-bu7s43-0 fywJMy"
    >
      Documentaries
    </p>
    <div
      class="card__Entities-bu7s43-12 gERdjH"
    >
      <div
        class="card__Item-bu7s43-13 gIHEnO"
        data-testid="amanda-knox-item-feature"
      >
        <img
          class="card__Image-bu7s43-14 kDZXcP"
          src="/images/series/documentaries/amanda-knox/small.jpg"
        />
        <div
          class="card__Meta-bu7s43-11 fwQvzH"
        >
          <p
            class="card__SubTitle-bu7s43-3 hNVrNW"
          >
            Amanda Knox
          </p>
          <p
            class="card__Text-bu7s43-4 gRjIdT"
          >
            Amanda Marie Knox is an American woman who spent almost four years in an Italian prison.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Card /> renders the <Card /> component with the populated data 1`] = `
<div
  class="card__Group-bu7s43-2 kghlzN"
>
  <div
    class="card__Container-bu7s43-1 gXEVLR"
  >
    <p
      class="card__Title-bu7s43-0 fywJMy"
    >
      Documentaries
    </p>
    <div
      class="card__Entities-bu7s43-12 gERdjH"
    >
      <div
        class="card__Item-bu7s43-13 gIHEnO"
      >
        <img
          class="card__Image-bu7s43-14 kDZXcP"
          src="/images/series/documentaries/amanda-knox/small.jpg"
        />
        <div
          class="card__Meta-bu7s43-11 fwQvzH"
        >
          <p
            class="card__SubTitle-bu7s43-3 hNVrNW"
          >
            Amanda Knox
          </p>
          <p
            class="card__Text-bu7s43-4 gRjIdT"
          >
            Amanda Marie Knox is an American woman who spent almost four years in an Italian prison.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
